ai-ml/ray/terraform/examples/pytorch/main.tf

```diff
@@ -0,0 +1,122 @@
+provider "aws" {
+  region = local.region
+}
+
+provider "kubernetes" {
+  host                   = data.aws_eks_cluster.this.endpoint
+  cluster_ca_certificate = base64decode(data.aws_eks_cluster.this.certificate_authority[0].data)
+
+  exec {
+    api_version = "client.authentication.k8s.io/v1beta1"
+    command     = "aws"
+    # This requires the awscli to be installed locally where Terraform is executed
+    args = ["eks", "get-token", "--cluster-name", local.eks_cluster]
+  }
+}
+
+provider "helm" {
+  kubernetes {
+    host                   = data.aws_eks_cluster.this.endpoint
+    cluster_ca_certificate = base64decode(data.aws_eks_cluster.this.certificate_authority[0].data)
+
+    exec {
+      api_version = "client.authentication.k8s.io/v1beta1"
+      command     = "aws"
+      # This requires the awscli to be installed locally where Terraform is executed
+      args = ["eks", "get-token", "--cluster-name", local.eks_cluster]
+    }
+  }
+}
+
+data "aws_eks_cluster" "this" {
+  name = local.eks_cluster
+}
+
+locals {
+  region      = "us-east-1"
+  name        = "pytorch"
+  eks_cluster = "ray-cluster"
+}
+
+module "pytorch_cluster" {
+  source = "../../modules/ray-cluster"
+
+  namespace        = local.name
+  ray_cluster_name = local.name
+  eks_cluster_name = local.eks_cluster
+
+  helm_values = [
+    yamlencode({
+      image = {
+        repository = "rayproject/ray-ml"
+        # This is a different version than the xgboost version
+        tag        = "2.3.0"
+        pullPolicy = "IfNotPresent"
+      }
```

Comment:
In my opinion, YAML is a big advantage for Kubernetes as it keeps things simple. It's much easier to work with, especially when editing configuration files. Many customers find it easier to update the YAML file directly with new configuration parameters, rather than having to modify Terraform code or learn how to convert YAML to Terraform. This simplicity and ease of use is what makes YAML a popular choice for Kubernetes configurations.